internal/types.go

```diff
@@ -8,3 +10,9 @@ type Receipt struct {
 	ThreadID  string `db:"thread_id"`
 	IsPrivate bool
 }
+
+type TimelineResponse struct {
```

Comment:
Isn't this just a `sync2.TimelineResponse`? Why not use that?

Reply:
Yes. There was some kind of import cycle when I tried to use this in more places. Let me see if I can dig the details up.

Reply:
If I move this back to sync2 and then try to build sync2 tests, I get a failure to build test binaries:

```
$ go list ./... | xargs -n1 go test -c -o /dev/null
?   	github.com/matrix-org/sliding-sync	[no test files]
?   	github.com/matrix-org/sliding-sync/cmd/syncv3	[no test files]
?   	github.com/matrix-org/sliding-sync/cmd/testmigration	[no test files]
?   	github.com/matrix-org/sliding-sync/pubsub	[no test files]
?   	github.com/matrix-org/sliding-sync/sqlutil	[no test files]
# github.com/matrix-org/sliding-sync/sync2
package github.com/matrix-org/sliding-sync/sync2
	imports github.com/matrix-org/sliding-sync/state: import cycle not allowed in test
FAIL	github.com/matrix-org/sliding-sync/sync2 [setup failed]
FAIL
?   	github.com/matrix-org/sliding-sync/testutils	[no test files]
?   	github.com/matrix-org/sliding-sync/testutils/m	[no test files]
```

Dep graph on main (`6f3f556`)

```
godepgraph --onlyprefixes github.com/matrix-org/sliding-sync github.com/matrix-org/sliding-sync | sed -e 's|github.com/matrix-org/sliding-sync/\?||g' | dot -Tpng -o godepgraph.png
```

![godepgraph](https://github.com/matrix-org/sliding-sync/assets/8614563/0ded6a58-9b41-4cbb-b609-68bb58b71f6b)

I don't understand where I've broken this. On `main` there _are_ references to sync2 inside the `state` tests, and to `state` inside the `sync2` code.


```
godepgraph -withtests -onlyprefixes github.com/matrix-org/sliding-sync github.com/matrix-org/sliding-sync | sed -e 's|github.com/matrix-org/sliding-sync/\?||g' | dot -Tpng -o godepgraph.png
```

![godepgraph](https://github.com/matrix-org/sliding-sync/assets/8614563/5d1f314f-6d58-41d9-a100-19e167c8b876)

Reply:
Removing the root package and `internal`:

```
godepgraph -onlyprefixes github.com/matrix-org/sliding-sync -i github.com/matrix-org/sliding-sync/internal github.com/matrix-org/sliding-sync | sed -e '/"github.com\/matrix-org\/sliding-sync"/d' -e 's|github.com/matrix-org/sliding-sync/||g' | dot -Tpng -o godepgraph.png
```

![godepgraph](https://github.com/matrix-org/sliding-sync/assets/8614563/ef62b6b8-3ce7-4de5-909f-5cdbe4952c8e)

Reply:
And even more readable:

```
godepgraph -onlyprefixes github.com/matrix-org/sliding-sync -i github.com/matrix-org/sliding-sync/internal github.com/matrix-org/sliding-sync | sed -e '/"github.com\/matrix-org\/sliding-sync"/d' -e 's|github.com/matrix-org/sliding-sync/||g' -e '/splines/d | dot -Tpng -o godepgraph.png
```

![godepgraph](https://github.com/matrix-org/sliding-sync/assets/8614563/3d767008-f7d3-4dc5-acc4-8256426313a2)

Reply:
The cycle was removed in #309.